src/main/java/edu/illinois/library/cantaloupe/processor/PdfBoxProcessor.java

```diff
@@ -187,6 +186,25 @@ public void put(COSObject indirect, PDXObject xobject) {
         }
     }
 
+    private MemoryUsageSetting getMemoryUsageSetting() {
+        final Configuration config = Configuration.getInstance();
+        final boolean scratchFileEnabled =
+                config.getBoolean(Key.PROCESSOR_PDF_SCRATCH_FILE_ENABLED, false);
+        final long maxMainMemoryBytes =
+                config.getLong(Key.PROCESSOR_PDF_MAX_MEMORY_BYTES, -1);
```

Comment:
For keys with byte-size values, it's better to use `Configuration.getLongBytes()` which enables users to enter a value with K/KB, M/MB, etc. suffix

Reply:
It seems as though for this you cannot have a `-1` value due to this line in `StringUtils`
```
final String numberStr = str.replaceAll("[^\\d.]", "");
```
It would need to be changed to this in order to allow `-1`
```
final String numberStr = str.replaceAll("[^-\\d.]", "");
```
I'm not sure what kind of side effect that may have (if any)

Reply:
That should be fine